UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/MetaDataLayer.cs

```diff
@@ -193,6 +193,16 @@ public void MakeSlipperyAt(Vector3Int position, bool canDryUp = true)
 	/// </summary>
 	public void ReagentReact(ReagentMix reagents, Vector3Int worldPosInt, Vector3Int localPosInt)
 	{
+		var mobs = MatrixManager.GetAt<LivingHealthMasterBase>(worldPosInt, true);
+		if (mobs is List<LivingHealthMasterBase>)
+		{
+			reagents.Divide((mobs as List<LivingHealthMasterBase>).Count + 1);
+			foreach (var mob in mobs)
+			{
+				mob.ApplyReagentsToSurface(reagents, BodyPartType.None);
+			}
+		}
```

Comment:
```suggestion
		reagents.Divide(mobs.Count() + 1);
		foreach (var mob in mobs)
		{
			mob.ApplyReagentsToSurface(reagents, BodyPartType.None);
		}
```

Reply:
the  if (mobs is List<LivingHealthMasterBase>) checks if it's not empty

Reply:
I get the idea, but you could get an empty list too.
You can just get the count and check that against 0, no need for casting.

Reply:
don't think you can get count on enumerator though?

Reply:
You can. It will call the Enumerable Count extension. It has some built in optimizations to check if the underlying collection is an ICollection and calls the ICollection's count. If it's not an ICollection, then it will enumerate through each item in the list to get the count. You might get a "possible multiple enumeration" warning in your IDE in either case though.

MatrixManager.GetAt returns an IEnumerable but it's actually a List (ICollection) so it should just return the list's count.

Reply:
"Did you look at the suggested code?" Just because you written it doesn't mean it compile : P

Reply:
![GIF164](https://user-images.githubusercontent.com/27101670/181201027-9cf5c968-e8c6-4df8-abac-277d8617e33d.gif)
 Ryder complains

Reply:
> Ryder complains

See my post above. "You might get a "possible multiple enumeration" warning in your IDE in either case though." 

Use Count, it should avoid actually enumerating twice and won't create garbage.

Reply:
> "Did you look at the suggested code?" Just because you written it doesn't mean it compile : P

Yeah, but I obviously checked that `Count()` exists.

The warning is not of concern in this context, as there's no expensive operation. You can either suppress the warning here, or return an `ICollection` from `MatrixManager.GetAt`. I'd just suppress it.

Reply:
possible multiple enumeration? what could that cause?

Reply:
You can technically have an `IEnumerable` that does a db query or similar. So it would be executed twice.
Shouldn't be relevant here though.